src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -87,12 +88,16 @@ mod tests;
 
 pub trait ForestObligation: Clone + Debug {
     type CacheKey: Clone + hash::Hash + Eq + Debug;
+    type Variable: Clone + hash::Hash + Eq + Debug;
+    type WatcherOffset;
```

Comment:
Ideally, we'd comment all these associated types too (one is pre-existing, obviously)